src/utils/verifiable-credential/snapVC.ts

```diff
@@ -0,0 +1,115 @@
+import type { Hex } from '@metamask/utils';
+
+import { BaseVerifiableCredential } from './baseVC';
+import {
+  TrustCredentialType,
+  SnapStatusReasonType,
+  SnapCurrentStatus,
+  type SnapDid,
+  type StatusReason,
+  type SnapCredentialSubject,
+} from './types';
+
+export class SnapVerifiableCredential extends BaseVerifiableCredential {
+  primaryType: TrustCredentialType = TrustCredentialType.StatusCredential;
+
+  credentialSubjectTypes = {
+    EIP712Domain: [
+      { name: 'name', type: 'string' },
+      { name: 'version', type: 'string' },
+      { name: 'chainId', type: 'uint256' },
+    ],
+    StatusReason: [
+      {
+        name: 'type',
+        type: 'string',
+      },
+      {
+        name: 'value',
+        type: 'string[]',
+      },
+    ],
+    CredentialSubject: [
+      {
+        name: 'id',
+        type: 'string',
+      },
+      {
+        name: 'currentStatus',
+        type: 'string',
+      },
+      {
+        name: 'statusReason',
+        type: 'StatusReason',
+      },
+    ],
+    StatusCredential: [
+      {
+        name: '@context',
+        type: 'string[]',
+      },
+      {
+        name: 'type',
+        type: 'string[]',
+      },
+      {
+        name: 'issuer',
+        type: 'string',
+      },
+      {
+        name: 'credentialSubject',
+        type: 'CredentialSubject',
+      },
+    ],
+  };
+
+  protected getSubjectDid(subjectAddress: string): SnapDid {
+    return `snap://${subjectAddress}`;
```

Comment:
Since I haven't seen this issue resolved in other places I'll ask here knowing this might be outside of scope for this PR.

Snap IDs use base64 with `/` in them. What's your way of handing slashes and discerning between base64 encoding and [base64url](https://en.wikipedia.org/wiki/Base64#Variants_summary_table) encoded snap checksum?

May I suggest using Opaque types from [utils](https://github.com/MetaMask/utils/blob/e97f2460d23ec9ed0907e5b68af8ecb84b28cb5a/src/opaque.ts#L4)

```typescript
declare const snapBase64Brand: unique symbol;
export type SnapBase64 = Opaque<string, snapBase64Brand>;

declare const snapBase64UrlBrand: unique symbol;
export type SnapBase64Url = Opaque<string, snapBase64UrlBrand>;
```

Reply:
thats even better, thank you

Reply:
May i ask how to use that SnapBase64? it doesnt look like valid for above definition, can we skip this at this PR?

Reply:
Exactly like that

Reply:
i merge this PR first, and will change back on that in other PR